x/tokenfactory/types/msgs.go

```diff
@@ -202,3 +204,43 @@ func (m MsgChangeAdmin) GetSigners() []sdk.AccAddress {
 	sender, _ := sdk.AccAddressFromBech32(m.Sender)
 	return []sdk.AccAddress{sender}
 }
+
+var _ sdk.Msg = &MsgSetDenomMetadata{}
+
+// NewMsgChangeAdmin creates a message to burn tokens
+func NewMsgSetDenomMetadata(sender string, metadata banktypes.Metadata) *MsgSetDenomMetadata {
+	return &MsgSetDenomMetadata{
+		Sender:   sender,
+		Metadata: metadata,
+	}
+}
+
+func (m MsgSetDenomMetadata) Route() string { return RouterKey }
+func (m MsgSetDenomMetadata) Type() string  { return TypeMsgSetDenomMetadata }
+func (m MsgSetDenomMetadata) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(m.Sender)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid sender address (%s)", err)
+	}
+
+	err = m.Metadata.Validate()
+	if err != nil {
+		return err
+	}
+
+	_, _, err = DeconstructDenom(m.Metadata.Base)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m MsgSetDenomMetadata) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&m))
+}
+
+func (m MsgSetDenomMetadata) GetSigners() []sdk.AccAddress {
+	sender, _ := sdk.AccAddressFromBech32(m.Sender)
```

Comment:
Can error be propagated to the caller?

Reply:
I don't think so.  This is part of the sdk.Msg interface, so we can't change the function signature